src/test/components/ListOfPublishedProfiles.test.js

```diff
@@ -284,54 +283,46 @@ describe('ListOfPublishedProfiles', () => {
   });
 
   describe('profile deletion', () => {
-    beforeEach(() => {
-      window.fetch = jest.fn();
-    });
-
-    afterEach(() => {
-      delete window.fetch;
-    });
-
     function mockFetchForDeleteProfile({
       endpointUrl,
       jwtToken,
     }: {
       endpointUrl: string,
       jwtToken: string,
     }) {
-      window.fetch.mockImplementation(async (urlString, options) => {
-        if (urlString !== endpointUrl) {
-          return new Response(null, { status: 404, statusText: 'Not found' });
-        }
-
-        const { method, headers } = options;
-
-        if (method !== 'DELETE') {
-          return new Response(null, {
-            status: 405,
-            statusText: 'Method not allowed',
+      window.fetch
+        .catch(404) // Catchall
+        .mock(endpointUrl, (urlString, options) => {
```

Comment:
This one is mostly similar to the previous code, due to https://github.com/wheresrhys/fetch-mock/issues/636